### PR TITLE
Makes emitters hitscan

### PR DESCRIPTION
**Pull request by NotRanged**
## About The Pull Request
image

Also fixes a visual bug with hitscan weaponry and reflector boxes.
Ports [(is this really a good idea) hitscan emitters #10450](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10450) with a few changes. Emitter muzzle and impact effect taken from baystation emitter sprites.

## Why It's Good For The Game

It looks better than slow moving laggy projectiles, and would be an interesting change of pace compared to the regular emitters.

## Changelog
:cl:
tweak: Emitters are now hitscan beams.
fix: Fixed a bug with hitscan beams and reflector boxes.
/:cl:

### CODE REVIEW

**Comment by Sparkezel** [comment]
Awesome for warcrimes, they wont know what hit them :ancap_thumbsup:

**Comment by MTandi** [comment]
I like the idea, but the beams look too thick. Maybe make them half the thickness?

**Comment by NotRanged** [comment]
Too thick? I actually considered making them a bit thicker and/or static-y/sparkly on the edges, as they seemed thin to me...

**Comment by Cheshify** [comment]
This'll make wacky reflector setups be easier to make, and easier to setup, and maybe people will walk straight into the lasers less often.

**Comment by CRITAWAKETS** [comment]
OH GOD

**Comment by NotRanged** [comment]
It'd look better if the beam segments would emit some light, but idk about lag concerns. Let's see...

**Comment by NotRanged** [comment]
image
Light does look kinda better yeah

**Comment by RaveRadbury** [comment]
How is damage applied with these now?

**Comment by NotRanged** [comment]
> How is damage applied with these now?

Same as before but without travel time. Like a hitscan weapon.

**Review by Twaticus — APPROVED** [review]

**Comment by zxaber** [comment]
Should have been done ages ago.

**Comment by ShibaInuLord** [comment]
how would this interact with blobs and reflective blob tiles?

**Comment by RaveRadbury** [comment]
> > How is damage applied with these now?
> 
> Same as before but without travel time. Like a hitscan weapon.

Ah, so they still fire in pulses, then?

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -690,14 +692,22 @@
...
 		var/matrix/M = new
 		M.Turn(Angle)
 		transform = M
 	if(trajectory)
 		trajectory.set_angle(new_angle)
+	if(fired && hitscan && isloc(loc) && (loc != last_angle_set_hitscan_store))
+		last_angle_set_hitscan_store = loc
+		var/datum/point/pcache = new (src)
+		var/list/L = list() 
+		L = trajectory.return_coordinates()
```
Comment (Jared-Fogle) [inline code/modules/projectiles/projectile.dm:707]: Better variable name. Also, why are you not just doing `L = trajectory.return_coordinates()`?

**Comment by necromanceranne** [comment]
> how would this interact with blobs and reflective blob tiles?

Instant reflection. So it'll look like nothing happened but the beam will have hit the reflector and then straight back the same direction, usually into the emitter.

**Comment by cacogen** [comment]
Sprites are bad but I like the idea. Further reminds me of laser puzzles in video games, like the one in Avernum 3.

File: code/modules/projectiles/projectile.dm (hunk 1 above)
Comment (NotRanged) [reply code/modules/projectiles/projectile.dm]: Fixed, thanks

**Pull request merged**